llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -17107,12 +17129,15 @@ SDValue DAGCombiner::visitFADD(SDNode *N) {
                            DAG.getConstantFP(4.0, DL, VT));
       }
     }
+  } // enable-unsafe-fp-math && AllowNewConst
 
+  if (((Options.UnsafeFPMath && Options.NoSignedZerosFPMath) ||
+       (Flags.hasAllowReassociation() && Flags.hasNoSignedZeros()))) {
```

Comment:
Does removing the `AllowNewConst` restriction from the reassociateReduction call have any effect on the tests. Just wondering if this is worth a separate PR?